Fail to run "mvn test" in MacOS

Hi, all:

While I ran "mvn test" on macOS, such exception occurred:

-------------------------------------------------------------------------------------

09:58:03.000|INFO |org.testcontainers.DockerClientFactory - Connected to docker:
  Server Version: 18.03.0-ce
  API Version: 1.37
  Operating System: Docker for Mac
  Total Memory: 1998 MB
09:58:04.135|INFO |org.testcontainers.DockerClientFactory - Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
09:59:24.361|ERROR|🐳 [gcr.io/etcd-development/etcd:v3.3] - Could not start container
java.lang.reflect.UndeclaredThrowableException: null
        at com.sun.proxy.$Proxy39.exec(Unknown Source) ~[?:?]
        at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:243) ~[testcontainers-1.10.5.jar:?]
        at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:216) ~[testcontainers-1.10.5.jar:?]
        at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:76) ~[duct-tape-1.0.7.jar:?]
        at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:214) ~[testcontainers-1.10.5.jar:?]
        at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:203) ~[testcontainers-1.10.5.jar:?]
        at io.etcd.jetcd.launcher.EtcdContainer.start(EtcdContainer.java:145) ~[classes/:?]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_151]
Caused by: java.lang.reflect.InvocationTargetException
        at sun.reflect.GeneratedMethodAccessor93.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_151]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_151]
        at org.testcontainers.dockerclient.AuditLoggingDockerClient.lambda$wrappedCommand$14(AuditLoggingDockerClient.java:98) ~[testcontainers-1.10.5.jar:?]
        ... 8 more
Caused by: com.github.dockerjava.api.exception.DockerException: Mounts denied:
The path /var/folders/l0/6jlf9plj197g06f2mjkv1h480000gn/T/jetcd_test_etcd0_1713473062908495631/
is not shared from OS X and is not known to Docker.
You can configure shared paths from Docker -> Preferences... -> File Sharing.
See https://docs.docker.com/docker-for-mac/osxfs/#namespaces for more info.

-------------------------------------------------------------------------------------

After investigation, I found it has sth to do with docker in macOS. The following question in stackoverflow shows exactly the same issue.
https://stackoverflow.com/questions/45122459/docker-mounts-denied-the-paths-are-not-shared-from-os-x-and-are-not-known

Since docker forbids to add /var/folder to its file sharing directories because /var/folder is actually a subdir of /private in macOS. My current solution is to reset environment variable "java.io.tmpdir" as follows:
`export _JAVA_OPTIONS=-Djava.io.tmpdir=/tmp/`

Is it the recommended solution to deal with such failure in macOS? Or maybe I should do some settings?

PS: It seems that testcontainer has adapted to macOS while create temporary directories. (Ref: org.testcontainers.utility#createTempDirectory())

--- a/jetcd-launcher/src/main/java/io/etcd/jetcd/launcher/EtcdContainer.java
+++ b/jetcd-launcher/src/main/java/io/etcd/jetcd/launcher/EtcdContainer.java
@@ -252,7 +252,10 @@ public class EtcdContainer implements AutoCloseable {
 
   private static Path createDataDirectory(String name) {
     try {
-      return Files.createTempDirectory("jetcd_test_" + name + "_");
+      final Path path = Files.createTempDirectory("jetcd_test_" + name + "_");
+      // https://github.com/etcd-io/jetcd/issues/489
+      // Resolve symlink (/var -> /private/var) to don't fail for Mac OS because of docker thing with /var/folders
+      return path.toRealPath();
     } catch (IOException e) {
       throw new ContainerLaunchException("Error creating data directory", e);
     }

